File: java/lsp-02-initial/src/test/java/no/bouvet/solid/lsp/RefundServiceTest.java
package no.bouvet.solid.lsp;

import static org.junit.Assert.assertTrue;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.services.PaymentType;
import no.bouvet.solid.lsp.services.RefundService;

import org.junit.Before;
import org.junit.Test;

public class RefundServiceTest {
    private RefundService subject;

    @Before
    public void setUp() {
        subject = new RefundService();
    }

    @Test
    public void testRefundPayPal() {
        // when
        Boolean result = subject.refund(PaymentType.PayPal, new BigDecimal(100), "ABC123");

        // then
        assertTrue(result);
    }

    @Test
    public void testRefundWorldPay() {
        // when
        Boolean result = subject.refund(PaymentType.WorldPay, new BigDecimal(100), "ABC123");

        // then
        assertTrue(result);
    }

}

File: dotnet/Solid/Tests/LSP/Example2/RefundServiceTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Solid.LSP.Example2.Services;

namespace Tests.LSP.Example2
{
    [TestClass]
    public class RefundServiceTest
    {
        private RefundService _subject;

        [TestInitialize]
        public void Setup()
        {
            _subject = new RefundService();
        }

        [TestMethod]
        public void TestRefundPayPal()
        {
            bool result = _subject.Refund(PaymentType.PayPal, 100, "ABC123");
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void TestRefundWorldPay()
        {
            bool result = _subject.Refund(PaymentType.WorldPay, 100, "ABC123");
            Assert.IsTrue(result);
        }
    }
}
